src/vm/memory/memory.zig

```diff
@@ -326,6 +342,24 @@ fn setUpMemory(memory: *Memory, comptime vals: anytype) !void {
     }
 }
 
+test "Memory: validate memory cell" {
```

Comment:
I think that we should add two more tests here:
- One to assert that if the segment index of the provided address in `validateMemoryCell` does not exist in `validation_rules`, then nothing change.
- One to assert that if the segment index of the provided address in `validateMemoryCell` exist in `validation_rules` but the address already exists in `validated_addresses` then nothing change.

Reply:
Gotcha. Also i have to finish the validate existing memory function and will add more tests to that as well.